ui/selectors/confirm-transaction.js

```diff
@@ -218,17 +228,46 @@ export const transactionFeeSelector = function (state, txData) {
   const currentCurrency = currentCurrencySelector(state);
   const conversionRate = conversionRateSelector(state);
   const nativeCurrency = getNativeCurrency(state);
+  const gasFeeEstimates = getGasFeeEstimates(state);
+  const gasEstimateType = getGasEstimateType(state);
 
-  const { txParams: { value = '0x0', gas: gasLimit = '0x0' } = {} } = txData;
-
-  // if the gas price from our infura endpoint is null or undefined
-  // use the metaswap average price estimation as a fallback
-  let { txParams: { gasPrice } = {} } = txData;
+  const gasEstimationObject = {
+    gasLimit: txData.txParams?.gas ?? '0x0',
+  };
 
-  if (!gasPrice) {
-    gasPrice = getAveragePriceEstimateInHexWEI(state) || '0x0';
+  switch (gasEstimateType) {
```

Comment:
this block is the best stab that shouldn't break anything for current txs. we need to think about the following variables:
TransactionType = 'legacy' | 'fee market'
gasEstiamteType = 'legacy' | 'eth_gasPrice' | 'none' | 'fee market'
doesGasPriceExistInTx = true | false
doesEIPGasFieldsExistOntx = true | false

Now create a table with what to do in each combinations of scenarios :P 

If the transaction type is FEE MARKET but our gas estimate type is ETH_gasPrice (because of a network failure or a dapp submits a type 0x2 tx on a non-supported network (we should fail at the provider level here but that's a todo item)). what do we do? 

when our gas estimates are FEE_MARKET and the dapp suggests gasPrice we just use the gasPrice and fall back to '0x0' for now.

😓

Reply:
After careful review and testing, I believe this code here is correct.